cpp/test/CMakeLists.txt

```diff
@@ -16,6 +16,7 @@
 
 # keep the files in alphabetical order!
 add_executable(test_raft
+    test/cluster/kmeans.cu
```

Comment:
Do we also need to move over the gather tests?

Reply:
True it should be tested as well. I moved the gather tests.